File: tmp/diagnose_routes.py
import os
import sys
import time
from playwright.sync_api import sync_playwright

output_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), 'review-screenshots'))
os.makedirs(output_dir, exist_ok=True)

routes = [
    '/login',
    '/',
    '/quiz',
    '/quiz/play',
    '/quiz/result',
    '/1v1',
    '/profile'
]

viewports = [
    {'name': 'mobile', 'width': 390, 'height': 844},
    {'name': 'desktop', 'width': 1440, 'height': 900}
]

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    
    for route in routes:
        print(f"\n==========================================")
        print(f"TESTING ROUTE: {route}")
        print(f"==========================================")
        
        context = browser.new_context(viewport={'width': 1440, 'height': 900})
        page = context.new_page()
        
        logs = []
        warnings = []
        errors = []
        failed_requests = []
        page_errors = []

        page.on("console", lambda msg: 
            errors.append(msg.text) if msg.type == "error" 
            else (warnings.append(msg.text) if msg.type == "warning" else logs.append(msg.text))
        )
        page.on("pageerror", lambda err: page_errors.append(str(err)))
        page.on("requestfailed", lambda req: failed_requests.append(f"{req.url} - {req.failure}"))

        url = f"http://localhost:4200{route}"
        try:
            res = page.goto(url, wait_until="networkidle", timeout=10000)
            status = res.status if res else "No response"
            final_url = page.url
            print(f"Initial status: {status}, Final URL: {final_url}")
        except Exception as e:
            print(f"Navigation exception: {e}")

        time.sleep(1)

        print(f"--- Console Errors ({len(errors)}) ---")
        for err in errors:
            print(f"  [ERROR] {err}")
        print(f"--- Console Warnings ({len(warnings)}) ---")
        for w in warnings:
            print(f"  [WARN] {w}")
        print(f"--- Page Errors ({len(page_errors)}) ---")
        for pe in page_errors:
            print(f"  [PAGE ERROR] {pe}")
        print(f"--- Failed Requests ({len(failed_requests)}) ---")
        for fr in failed_requests:
            print(f"  [FAILED REQ] {fr}")

        context.close()

    browser.close()

File: tmp/diagnose_routes.js
const { chromium } = require('playwright-core');

const executablePath = 'C:\\Users\\Abolfazl\\AppData\\Local\\ms-playwright\\chromium-1243\\chrome-win64\\chrome.exe';

const routes = [
  '/login',
  '/',
  '/quiz',
  '/quiz/play',
  '/quiz/result',
  '/1v1',
  '/profile'
];

(async () => {
  const browser = await chromium.launch({ executablePath, headless: true });
  const report = {};

  for (const route of routes) {
    const context = await browser.newContext({ viewport: { width: 1440, height: 900 } });
    const page = await context.newPage();

    const routeData = {
      initialRoute: route,
      finalUrl: null,
      status: null,
      consoleErrors: [],
      consoleWarnings: [],
      consoleLogs: [],
      pageErrors: [],
      failedRequests: []
    };

    page.on('console', msg => {
      const type = msg.type();
      const text = msg.text();
      if (type === 'error') routeData.consoleErrors.push(text);
      else if (type === 'warning') routeData.consoleWarnings.push(text);
      else routeData.consoleLogs.push(text);
    });

    page.on('pageerror', err => {
      routeData.pageErrors.push(err.toString());
    });

    page.on('requestfailed', req => {
      routeData.failedRequests.push(`${req.url()} - ${req.failure()?.errorText || 'failed'}`);
    });

    const url = `http://localhost:4200${route}`;
    try {
      const response = await page.goto(url, { waitUntil: 'domcontentloaded', timeout: 10000 });
      routeData.status = response ? response.status() : 'No response';
      await page.waitForTimeout(1500);
      routeData.finalUrl = page.url();
    } catch (e) {
      routeData.error = e.toString();
    }

    report[route] = routeData;
    await context.close();
  }

  await browser.close();
  console.log(JSON.stringify(report, null, 2));
})();
